plugins/modules/perlmod_install_info.py

```diff
@@ -0,0 +1,410 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023, Jonathan Kamens <jik@kamens.us>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# pylint: disable=missing-module-docstring
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: perlmod_install_info
+
+short_description: Determine from where to install Perl modules
+
+requirements:
+    - C(apt-file) executable in search path on systems that use the apt package
+      manager
+    - C(dnf) or C(yum) executable in search path on systems that use the
+      dnf/yum package manager
+    - C(cpanm) executable in search path if you want to be able to search for
+      packages using cpanm
+
+version_added: "7.0.0"
+
+description:
+    - Searches dnf, yum, apt, and/or cpanm to determine the best source from
+      which to install a Perl module.
+    - Prefers the OS repositories over cpanm. Only returns modules from CPAN
+      when it is searchable (I(try_cpan) is C(auto) or C(true) and the C(cpanm)
+      executable is available) and the returned modules were not found
+      in any other source.
+    - Specify module names as you would specify them with the C(use) command in
+      a Perl script.
+    - Does not actually install modules. Instead, returns information about
+      where they can be installed from, which can be supplied to subsequent
+      tasks to do the actual installation.
+    - Note that this module will not fail by default if it cannot locate a
+      requested module. If you want that behavior, include a C(failed_when)
+      which checks for C(missing) being defined as shown in the examples.
+
+options:
+    name:
+        description: Specify one or more Perl modules to search for.
+        required: true
+        type: list
+        elements: str
+    try_installed:
+        description: Specify whether to check if modules are already installed
+          and not look elsewhere if they are.
+        type: bool
+        default: true
+    try_dnf:
+        description: Specify whether to check for modules using the dnf package
+          manager. Defaults to C(true) if C(dnf) executable is available.
```

Comment:
Please use semantic markup as documented in:
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation
```suggestion
          manager. Defaults to V(true) if C(dnf) executable is available.
```
Please review the rest of the docs.